### ISSUE TITLE
task: create a container for terminals spawned by tasks

### ISSUE THREAD

**Issue by vince-fugnitto** [eclipse-theia/theia#7274]
**Description**

The **proposal** is to create a new container for terminals spawned by tasks similarly to how the _output_ view displays multiple channels in a dropdown. User triggered integrated terminals can still work like today.

After #7260, we can enhance terminals created by tasks to be located in a common location within a widget container. Since these terminals cannot be re-used by users and only tasks, having a common location for all tasks looks to be a good option.

It can look something like the following:

![image](https://user-images.githubusercontent.com/40359487/75895325-793a5400-5e03-11ea-8acb-84a2f8948ae9.png)

cc @elaihau

**Comment by akosyakov** [comment]
cc @svenefftinge

**Comment by akosyakov** [comment]
I don't think we want it, it is nice that one can change layout of terminal for different tasks. Maybe a quick open instead to look up a terminal?

**Comment by vince-fugnitto** [comment]
> I don't think we want it

I was thinking users might appreciate to have a centralized locations for their `tasks`, and be able to view them easily without the clutter or noise of opening in different locations. Adding a `preference` to control this behavior (to have a container or not) might be too complex if we were to support both approaches.

> Maybe a quick open instead to look up a terminal?

I believe we do have the quick open to view terminals today.

**Comment by akosyakov** [comment]
> Adding a preference to control this behavior (to have a container or not) might be too complex if we were to support both approaches.

Is not having 2 approaches add complexity from user and implementation perspectives?

> I believe we do have the quick open to view terminals today.

Could it be used to filter terminals by task id or label?

**Comment by vince-fugnitto** [comment]
> > Adding a preference to control this behavior (to have a container or not) might be too complex if we were to support both approaches.
> 
> Is not having 2 approaches add complexity from user and implementation perspectives?

Right, I think it'd be too complex to support both the current approach and the containers.

> > I believe we do have the quick open to view terminals today.
> 
> Could it be used to filter terminals by task id or label?

Yes we definitely can:

<div align='center'>

![image](https://user-images.githubusercontent.com/40359487/75915932-6e90b680-5e25-11ea-8922-03880445ee4d.png)

</div>
